Added a timeout transport

chrisbeckey opened the pull request:
I have an unusual test case where I need to simulate a timeout from SMTP on a thread that itself has a timeout. Adding a TimeoutTransport to your library was the simplest way for me to do that. Hope it is useful.
BTW, I needed to add <tarLongFileMode>posix<tarLongFileMode> to the maven-assembly-plugin in the POM, it was failing to build on OSX 10.14.6.

ceharris commented:
Nice work and awesome new feature!

Could I convince you to add just a few words about how to use it in the README?

chrisbeckey commented:
Added to the README. I also changed the TimeoutTransport to derive from FileTransport because the timeouts can both be set to -1, in which case TimeoutTransport becomes a passthrough and it should have something to passthrough to.

chrisbeckey commented:
Done, and added a comment on the PR to explain why I derived
TimeoutTransport from FileTransport.


On Tue, Oct 27, 2020 at 11:20 AM Carl Harris <notifications@github.com>
wrote:

> *This email was sent from an external source so please treat with caution.*
> ------------------------------
> Nice work and awesome new feature!
>
> Could I convince you to add just a few words about how to use it in the
> README?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/soulwing/javashmail-provider/pull/3#issuecomment-717316286>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMYZJ7E7BPPBMFTTNLXSJ4LSM3QDJANCNFSM4TA7A5EQ>
> .
>

About Ascential plc: Ascential (LSE:ASCL.L) is a specialist information, data and analytics company that helps the world’s most ambitious businesses win in the digital economy. Our information, insights, connections, data and digital tools solve customer problems in three disciplines: 
Product Design via global trend forecasting service WGSN;
Marketing via global benchmark for creative excellence and effectiveness Cannes Lions and WARC and strategic advisory firm MediaLink; and 
Sales via ecommerce-driven data, insights and advisory service Edge by Ascential, leading managed services provider for Amazon Flywheel Digital, the world's premier payments and Fin Tech congress Money20/20, global retail industry summit World Retail Congress and retail news outlet Retail Week.
Ascential also powers political, construction and environmental intelligence brands DeHavilland, Glenigan and Groundsure.
The information in or attached to this email is confidential and may be legally privileged. If you are not the intended recipient of this message, any use, disclosure, copying, distribution or any action taken in reliance on it is prohibited and may be unlawful. 
If you have received this message in error, please notify the sender immediately by return email and delete this message and any copies from your computer and network. Ascential does not warrant that this email and any attachments are free from viruses and accepts no liability for any loss resulting from infected email transmissions.
Ascential reserves the right to monitor all email through its networks. Any view expressed may be those of the originator and not necessarily of Ascential plc. 
Please be advised all phone calls may be recorded for training and quality purposes and by accepting and/or making calls to us you acknowledge and agree to calls being recorded. 
Ascential plc, number 9934451 (England and Wales). Registered Office: The Prow, 1 Wilder Walk, London W1B 5AP.

ceharris commented:
I have a better solution for the need to delegate to another transport, that I had been meaning to generalize.

I created a `DelegatingTransport` that any transport can extend, which uses properties like these to create a delegate session and transport.

```
mail.transport.protocol=timeout
mail.timeout.delegate.transport.protocol=file
mail.timeout.delegate.file.path=/some/path/to/the/file
```

Basically, you just insert `<yourprotocol>.delegate` in between `mail.` and `.some.property` for each property of the delegate transport.

Rather than trouble you to rebase your pull request, I'll merge it as is, and then refactor it to make use of the new `DelegatingTransport` and update the README accordingly.

chrisbeckey commented:
Cool, I prefer your delegate approach as much more general usage. Thanks
for writing the library.

On Tue, Oct 27, 2020 at 1:07 PM Carl Harris <notifications@github.com>
wrote:

> *This email was sent from an external source so please treat with caution.*
> ------------------------------
> Merged #3
> <https://github.com/soulwing/javashmail-provider/pull/3>
> into master.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/soulwing/javashmail-provider/pull/3#event-3927118681>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMYZJ7GTH5UTVYYJOSB2ULTSM345JANCNFSM4TA7A5EQ>
> .
>

About Ascential plc: Ascential (LSE:ASCL.L) is a specialist information, data and analytics company that helps the world’s most ambitious businesses win in the digital economy. Our information, insights, connections, data and digital tools solve customer problems in three disciplines: 
Product Design via global trend forecasting service WGSN;
Marketing via global benchmark for creative excellence and effectiveness Cannes Lions and WARC and strategic advisory firm MediaLink; and 
Sales via ecommerce-driven data, insights and advisory service Edge by Ascential, leading managed services provider for Amazon Flywheel Digital, the world's premier payments and Fin Tech congress Money20/20, global retail industry summit World Retail Congress and retail news outlet Retail Week.
Ascential also powers political, construction and environmental intelligence brands DeHavilland, Glenigan and Groundsure.
The information in or attached to this email is confidential and may be legally privileged. If you are not the intended recipient of this message, any use, disclosure, copying, distribution or any action taken in reliance on it is prohibited and may be unlawful. 
If you have received this message in error, please notify the sender immediately by return email and delete this message and any copies from your computer and network. Ascential does not warrant that this email and any attachments are free from viruses and accepts no liability for any loss resulting from infected email transmissions.
Ascential reserves the right to monitor all email through its networks. Any view expressed may be those of the originator and not necessarily of Ascential plc. 
Please be advised all phone calls may be recorded for training and quality purposes and by accepting and/or making calls to us you acknowledge and agree to calls being recorded. 
Ascential plc, number 9934451 (England and Wales). Registered Office: The Prow, 1 Wilder Walk, London W1B 5AP.

ceharris commented:
I have merged and refactored your contribution to use the new `DelegatingTransport` on latest master. I also cut a new a release which should hit Maven Central indexes in the next couple of hours.
